### PR TITLE
feat(internal/config): add additional_protos configuration support for php

### PR DESCRIPTION
**Pull request by zhumin8**
Add support for `additional_protos` in PHP package and API-level configuration structures. This enables specifying mixin proto dependencies (e.g., location.proto, iam_policy.proto) to compile them directly as target protos for client code generation.

For #6743

### CODE REVIEW

File: internal/librarian/library_test.go [hunk 1]
```diff
@@ -1566,3 +1566,38 @@ func TestMergeRust(t *testing.T) {
...
+			want: &config.PHPPackage{AdditionalProtos: []string{"a.proto"}},
+		},
+		{
+			name: "nil dst returns src",
+			dst:  nil,
+			src:  &config.PHPPackage{AdditionalProtos: []string{"b.proto"}},
+			want: &config.PHPPackage{AdditionalProtos: []string{"b.proto"}},
+		},
+		{
+			name: "merges all fields",
```
Comment (JoeWang1127) [inline internal/librarian/library_test.go:1590]: Could you add a test case to verify additional protos in the merging result has no duplicates?

**Review by JoeWang1127 — APPROVED** [review]

**Comment by JoeWang1127** [comment]
I think the test should be fine if you merge main branch.

File: internal/librarian/library_test.go (hunk 1 above)
Comment (zhumin8) [reply internal/librarian/library_test.go:1590]: It looks like that the "merging" in  ResolvePreview  is primarily designed as an override mechanism (preview replaces stable values), rather than a combining mechanism (union). So I don't think this is a good place to dedup.
However, src.AdditionalProtos  itself might contain duplicates and I will create a followup to Tidy to validate/dedup.

In the meantime, it looks like the merge functions for each language is a bit inconsistent and I am not super clear about what needs to be merged and what not. I will create a separate issue for clarifying and perhaps refactor for each language.

Comment (zhumin8) [reply internal/librarian/library_test.go:1590]: Filed https://github.com/googleapis/librarian/issues/6809 for ResolvePreview
And added note to https://github.com/googleapis/librarian/issues/6743#issuecomment-4970438762
